### PR TITLE
Add ExpGrad registration and test.

### PR DESCRIPTION
**Pull request by mrry**
**Description**: Add missing gradient registration for the `Exp` op.

**Motivation and Context**
- Adding support for training a model that uses the `Exp` op.

### CODE REVIEW

File: orttraining/orttraining/test/gradient/gradient_ops_test.cc [hunk 1]
```diff
@@ -416,6 +416,10 @@ TEST(GradientCheckerTest, LogGrad) {
   EXPECT_IS_TINIER_THAN(max_error, error_tolerance);
 }
 
+TEST(GradientCheckerTest, ExpGrad) {
+  UnaryOpGradientTest("Neg");
```
Comment (SherlockNoMad) [inline orttraining/orttraining/test/gradient/gradient_ops_test.cc:420]: "Exp"

Comment (mrry) [reply orttraining/orttraining/test/gradient/gradient_ops_test.cc:420]: Ha, thanks for catching that! Seemed too easy :P.

**Comment by SherlockNoMad** [comment]
> @SherlockNoMad I added a gradient checker test for the new registration - is that sufficient or are there other tests that would be useful?

For this simple gradient, I think it's sufficient. :) 
For other gradients ops that involves a hand-written kernel, we would need a better coverage. see layer_norm_test.cc as an example.

**Comment by mrry** [comment]
Thanks Sherlock! It turned out that actually testing the code revealed some issues :). In particular, I had to narrow the distribution for random values of `x` so that it doesn't generate too-large values that strain the fp32 precision. Clamping it to `[-1.0, 1.0]` seems to test out fine though.

**Review by SherlockNoMad — APPROVED** [review]

**Pull request merged**